src/plugincollection.js

```diff
@@ -86,83 +86,114 @@ export default class PluginCollection {
 	 *  which should not be loaded (despite being specified in the `plugins` array).
 	 * @returns {Promise} A promise which gets resolved once all plugins are loaded and available into the
 	 * collection.
-	 * @returns {Array.<module:core/plugin~Plugin>} returns.loadedPlugins The array of loaded plugins.
+	 * @returns {Promise.<Array.<module:core/plugin~Plugin>>} returns.loadedPlugins The array of loaded plugins.
 	 */
 	load( plugins, removePlugins = [] ) {
 		const that = this;
 		const editor = this._editor;
 		const loading = new Set();
 		const loaded = [];
 
-		// Plugins that will be removed can be the constructors or names.
-		// We need to unify this because we are supporting loading plugins using both types.
-		removePlugins = removePlugins.reduce( ( arr, PluginConstructorOrName ) => {
-			arr.push( PluginConstructorOrName );
-
-			if ( typeof PluginConstructorOrName == 'string' ) {
-				arr.push( getPluginConstructor( PluginConstructorOrName ) );
-			} else if ( PluginConstructorOrName.pluginName ) {
-				arr.push( PluginConstructorOrName.pluginName );
-			}
+		// In order to avoid using plugin names or constructors alternatively, we map all passed plugin as plugins constructors.
+		const pluginConstructorsToLoad = plugins
+			.map( ( item ) => getPluginConstructor( item ) )
+			.filter( ( PluginConstructor ) => !!PluginConstructor );
+
+		// Check whether a length of passed array with plugins is the same like an array with mapped plugins.
```

Comment:
Besides, I don't like splitting the filtering  which is done above from the check which is done below because it's a single piece of logic – can't find a plugin – throw.

OTOH, I can see that when using `filter()` it would be hard to reject the promise and stop the execution, so the check makes sense to be separated. So let's keep it like that, but please move it to a separate function. If you'll name this function intuitively, you won't need that comment :P.

The code in here could look like this:

```js
const missingPlugins = getMissingPlugins( plugins, pluginConstructorsToLoad );

if ( missingPlugins ) {
    return Promise.reject();
}
```